### DISCUSSION TITLE
Timeout for o1-preview model

### DISCUSSION THREAD

**Discussion by david-strejc** [danny-avila/LibreChat#4111] [Troubleshooting]
### What happened?

Frontend is reporting timeout - the chat name is not updated, bud after waiting without refresh there is response in the "New Chat" present.

### Steps to Reproduce

o1-preview model with longer response time 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Not enough information to reproduce and cannot reproduce.

**Reply by david-strejc** [reply]
It is harder to reproduce but I will describe it - I post a long lasting question to the o1-preview - response takes more than 1 minute for sure which is normal for o1 - than on frontend there is "no response from backend please refres the page" - then if I refresh the page response is there so backend received response OK. But the naming of chat is not updated in this case.

**Reply by danny-avila** [reply]
This is affected by your nginx/traefik/reverse-proxy config, which may timeout connnections that take long to response. Are you using a service like that?

**Reply by david-strejc** [reply]
> This is affected by your nginx/traefik/reverse-proxy config, which may timeout connnections that take long to response. Are you using a service like that?

MANY thanks - my fault - there was 60s and I didn't realize this can cause the timeout bacause frontend is already loaded and there are javascript requests.

**Reply by thrillhousedev** [reply]
I wanted to add for others facing this issue. We deployed the application to Azure and are leveraging an application gateway. To resolve the timeout issues, I had to both increase the timeout within the NGINX config, but also had to increase it within the app gateway's backend settings. I was going mad because of 20 second timesouts after I increased the timeouts to 300 seconds within NGINX, realizing the app gateway was at 20 seconds. Problem solved and now we can generate DALL-E and o1 responses without timeouts.

**Comment by vlad-tsoy** [comment]
@thrillhousedev, would you be willing to share the updated files? I'm facing a similar issue.
